micropython/upysh/upysh.py

```diff
@@ -1,31 +1,41 @@
-import sys
+try:
+    import usys as sys
+except:
+    import sys
 import os
 
-
 class LS:
+
     def __repr__(self):
         self.__call__()
         return ""
 
     def __call__(self, path="."):
-        l = os.listdir(path)
+        l = list(os.ilistdir(path))
         l.sort()
         for f in l:
-            st = os.stat("%s/%s" % (path, f))
-            if st[0] & 0x4000:  # stat.S_IFDIR
-                print("   <dir> %s" % f)
-            else:
-                print("% 8d %s" % (st[6], f))
-
+            if f[1] == 0x4000:  # stat.S_IFDIR
+                print("   <dir> %s" % f[0])
+        for f in l:
+            if f[1] != 0x4000:
+                if len(f) > 3:
+                    print("% 8d %s" % (f[3], f[0]))
+                else:
+                    print("         %s" % f[0])
+        try:
+            st = os.statvfs(path)
+            print("\n{:,d}k free".format(st[1] * st[3] // 1024))
+        except:
+            pass
 
 class PWD:
+
```

Comment:
Unless black wants these lines added, I'd rather not add them.

Reply:
OK. I'll ask Black.